src/Tests/Microsoft.NET.Sdk.Razor.Tests/BuildIntegrationTest.cs

```diff
@@ -0,0 +1,856 @@
+        }
+
+        [Fact]
+        public void Build_DoesNotAddRelatedAssemblyPart_IfViewCompilationIsDisabled()
+        {
+            var testAsset = "SimpleMvc";
+            var projectDirectory = _testAssetsManager
+                .CopyTestAsset(testAsset)
+                .WithSource();
+            
+            var build = new BuildCommand(projectDirectory);
+            build.Execute("/p:RazorCompileOnBuild=false", "/p:RazorCompileOnPublish=false").Should().Pass();
+
+            var intermediateOutputPath = build.GetIntermediateDirectory("net5.0", "Debug").ToString();
+            var razorAssemblyInfo = Path.Combine(intermediateOutputPath, "SimpleMvc.RazorAssemblyInfo.cs");
+
+            new FileInfo(razorAssemblyInfo).Should().NotExist();
+            new FileInfo(Path.Combine(intermediateOutputPath, "SimpleMvc.RazorTargetAssemblyInfo.cs")).Should().NotExist();
+        }
+
+        [Fact]
+        public void Build_AddsRelatedAssemblyPart_IfCompileOnPublishIsAllowed()
+        {
+            var testAsset = "SimpleMvc";
+            var projectDirectory = _testAssetsManager
+                .CopyTestAsset(testAsset)
+                .WithSource();
+            
+            var build = new BuildCommand(projectDirectory);
+            build.Execute("/p:RazorCompileOnBuild=false").Should().Pass();
+
+            var intermediateOutputPath = build.GetIntermediateDirectory("net5.0", "Debug").ToString();
+            var razorAssemblyInfo = Path.Combine(intermediateOutputPath, "SimpleMvc.RazorAssemblyInfo.cs");
+
+            new FileInfo(razorAssemblyInfo).Should().Exist();
+            new FileInfo(razorAssemblyInfo).Should().Contain("[assembly: Microsoft.AspNetCore.Mvc.ApplicationParts.RelatedAssemblyAttribute(\"SimpleMvc.Views\")]");
+
+            new FileInfo(Path.Combine(intermediateOutputPath, "SimpleMvc.RazorTargetAssemblyInfo.cs")).Should().NotExist();
+        }
+
+        [Fact]
+        public void Build_AddsRelatedAssemblyPart_IfGenerateAssemblyInfoIsFalse()
+        {
+            var testAsset = "SimpleMvc";
+            var projectDirectory = _testAssetsManager
+                .CopyTestAsset(testAsset)
+                .WithSource();
+            
+            var build = new BuildCommand(projectDirectory);
+            build.Execute("/p:GenerateAssemblyInfo=false").Should().Pass();
+
+            var intermediateOutputPath = build.GetIntermediateDirectory("net5.0", "Debug").ToString();
+            var razorAssemblyInfo = Path.Combine(intermediateOutputPath, "SimpleMvc.RazorAssemblyInfo.cs");
+
+            new FileInfo(razorAssemblyInfo).Should().Exist();
+            new FileInfo(razorAssemblyInfo).Should().Contain("[assembly: Microsoft.AspNetCore.Mvc.ApplicationParts.RelatedAssemblyAttribute(\"SimpleMvc.Views\")]");
+        }
+
+        [Fact]
+        public void Build_WithGenerateRazorHostingAssemblyInfoFalse_DoesNotGenerateHostingAttributes()
```

Comment:
Remove the hosting assembly info tests if you'd like. Nothing uses these attributes, so we don't super care about these any longer.